fields/types/gcsavatar/GcsAvatarField.js

```diff
@@ -0,0 +1,147 @@
+import Field from '../Field';
+import React from 'react';
+import ReactDOM from 'react-dom';
+import get from 'lodash/get';
+import { Button, FormField, FormNote } from 'elemental';
+
+const _ = {
+  get,
+};
+
+const storageDefaults = {
+  scheme: 'https',
+  hostname: 'storage.googleapis.com',
+};
+
+const supportTypes = ['image/gif', 'image/png', 'image/jpeg', 'image/x-tiff', 'image/svg+xml', 'image/bmp'];
+
+module.exports = Field.create({
+  displayName: 'GcsAvatarField',
+  getThumbnail() {
+    const { scheme, hostname } = storageDefaults;
+    const gcsBucket = _.get(this.props, 'value.gcsBucket', '');
+    const gcsDir = _.get(this.props, 'value.gcsDir', '');
+    const filename = _.get(this.props, 'value.filename', '');
+    if (gcsBucket && gcsDir && filename) {
+      return `${scheme}://${hostname}/${gcsBucket}/${gcsDir}${filename}`;
+    }
+  },
+  /**
+   *  is this image uploaded from end user local workspace in this session?
+   */
+  hasLocal() {
+    return this.state.origin === 'local';
+  },
+  getImageSource() {
+    if (this.hasLocal()) {
+      return this.state.localSource;
+    } else if (this.getThumbnail()) {
+      return this.getThumbnail();
+    } else {
+      return null;
+    }
+  },
+  /**
+   * Render an image preview
+   */
+  renderImagePreview() {
+    const thumbnail = this.getImageSource();
+
+    if (thumbnail) {
+      return (
+        <div className="image-preview">
+          <a className="img-thumbnail" href={thumbnail} target="_blank">
+            <img className="img-load" style={{ height: '90' }} src={thumbnail} alt="圖片無法載入，請稍後再試。"/>
```

Comment:
our avatar size is 82 * 82 ?

Reply:
Do you mean the rendered size of [AvatarImg](https://github.com/twreporter/keystone-plugin/blob/f5e5f27453ea67ab4055edf9777ae3740f960d0c/src/client/bundle/components/avatar.js#L41) ?

I think it's ok to leave it as 90px here since we only assign the height value and the width would be auto.

It is the screenshot:
![image](https://user-images.githubusercontent.com/6375655/132670340-732727d4-f03d-4bc6-953b-a7419b312517.png)

Reply:
OK!